tests/unit/processor/generic_adder/test_generic_adder.py

```diff
@@ -270,13 +296,41 @@ class TestGenericAdderProcessorSQLWithoutAddedTarget(BaseProcessorTestCase):
         },
     }
 
-    @property
-    def generic_rules_dirs(self):
-        return self.CONFIG.get("generic_rules")
+    def test_load_from_file(self):
+        expected_db = {
+            "TEST_0": (["b", "foo"], ["c", "bar"]),
+            "TEST_1": (["b", "uuu"], ["c", "vvv"]),
+            "TEST_2": (["b", "123"], ["c", "456"]),
+        }
 
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG.get("specific_rules")
+        assert self.object._db_table == expected_db
+
+        _, temp_path = tempfile.mkstemp()
+        self.object._db_file_path = temp_path
+
+        db_file_content = {"FOO": "BAR"}
+        with open(temp_path, "w", encoding="utf8") as db_table:
+            json.dump(db_file_content, db_table)
+        self.object._load_from_file()
+
+        assert self.object._db_table == db_file_content
+
+    def test_check_if_file_not_exists_or_stale(self):
+        db_file_path = self.object._db_file_path
+
+        stale_before_interval = self.object._check_if_file_not_exists_or_stale(time.time())
+        time.sleep(0.2)  # nosemgrep
+        now = time.time()
+        stale_after_interval = self.object._check_if_file_not_exists_or_stale(now)
+        os.utime(db_file_path, (now, now))
+        stale_after_change = self.object._check_if_file_not_exists_or_stale(now)
+        os.remove(db_file_path)
+        stale_without_file = self.object._check_if_file_not_exists_or_stale(now)
+
+        assert stale_before_interval is False
+        assert stale_after_interval is True
+        assert stale_after_change is False
+        assert stale_without_file is True
```

Comment:
In my opinion, this test is not understandable at first glance which might warrant a doctoring even though it is a test.

Reply:
I have split it into smaller tests so that it is easier to understand it.